File: 7.5 SharedPrefApp/app/src/main/java/com/example/sharedprefapp/MainActivity.java
package com.example.sharedprefapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.TextView;



public class MainActivity extends AppCompatActivity {


    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);

        SharedPreferences sharedPreferences = getSharedPreferences("MyPref",MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putString("name","vikas");
        editor.commit();

        String s = sharedPreferences.getString("name","");
        textView.setText(s);



    }
}

File: __Kotlin Apps/SharedPrefApp/app/src/main/java/com/example/sharedprefapp/MainActivity.kt
package com.example.sharedprefapp

import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

     lateinit var textView:TextView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val button : Button = findViewById(R.id.button)
        val editText : EditText = findViewById(R.id.editTextText)

        textView = findViewById(R.id.textView)

        displayDataFromSharedPref()


        button.setOnClickListener(View.OnClickListener {

            val enteredText = editText.text.toString()
            saveToSharedPref(enteredText)

        })




    }
    private fun saveToSharedPref (enteredText:String){
        val sharedPreferences:SharedPreferences = getSharedPreferences("mySharedPref", MODE_PRIVATE)
        val sharedPrefEditor:SharedPreferences.Editor = sharedPreferences.edit()
        sharedPrefEditor.putString("name",enteredText).apply()
    }

    private fun displayDataFromSharedPref (){
        val sharedPreferences:SharedPreferences = getSharedPreferences("mySharedPref", MODE_PRIVATE)
        val name = sharedPreferences.getString("name","")
        textView.text = name
    }
}

